File: Python27/project_euler/001_050/003.py
def prob_003(num):
    '''Returns the largest prime factor of a number'''
    i = 3
    while i <= num:
        if num % i:
            pass
        else:
            ans = i
            num /= i
            while True:
                if num % i:
                    break
                else:
                    num /= i

        i += 2

    return ans

if __name__ == "__main__":
    print prob_003(600851475143)

File: C/project_euler/001_050/003.c
/**
The prime factors of 13195 are 5, 7, 13 and 29.
What is the largest prime factor of the number 600851475143 ?
*/
#include<stdio.h>
#include<stdlib.h>

int main()
{
    long long int NUM = 600851475143;
    int prime = 3;

    for (long long int i = prime; i <= NUM; i += 2)
    {
        while (NUM % i == 0)
        {
            prime = i;
            NUM = NUM / i;
        }
    }
    printf("%d", prime);

    return 0;
}
